packages/core/utils/src/__tests__/content-types.test.ts

```diff
@@ -110,6 +111,30 @@ describe('Content types utils', () => {
     });
   });
 
+  describe('getDoesAttributeRequireValidation', () => {
+    test('false for field without constraints', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'string' })).toEqual(false);
+    });
+    test('true for required fields', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'string', required: true })).toEqual(true);
+    });
+    test('true for unique fields', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'string', unique: true })).toEqual(true);
+    });
+    test('true for numeric fields with a max', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'integer', max: 10 })).toEqual(true);
+    });
+    test('true for numeric fields with a min', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'integer', min: 10 })).toEqual(true);
+    });
+    test('true for fields with a max length', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'string', maxLength: 10 })).toEqual(true);
+    });
+    test('true for fields with a min length', () => {
+      expect(getDoesAttributeRequireValidation({ type: 'string', minLength: 10 })).toEqual(true);
+    });
```

Comment:
Maybe also test the falsy versions of each of these should return false (for required and unique at least)

Reply:
Done here [`e2c5b0c` (#20046)](https://github.com/strapi/strapi/pull/20046/commits/e2c5b0c6a6a04d0d6a2a685057798d85c89bde96)